pkg/api/validate/dynamic/dynamic.go

```diff
@@ -590,34 +598,75 @@ func (dv *dynamic) ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster
 			)
 		}
 
-		if oc.Properties.ProvisioningState == api.ProvisioningStateCreating {
-			if ss.SubnetPropertiesFormat != nil &&
-				ss.SubnetPropertiesFormat.NetworkSecurityGroup != nil {
-				expectedNsgID, err := subnet.NetworkSecurityGroupID(oc, s.ID)
+		subnetByID[s.ID] = ss
+	}
+	return subnetByID, nil
+}
+
+// checkByoNSG checks whether all the subnets have or don't have NSG attached.
+// when the BYONsg feature flag is on and only some of the subnets are attached with an NSG,
+// it returns an error.  If none of the subnets is attached, it's no longer BYO NSG and the
+// cluster installation process should fall back to using the managed nsg.
+func (dv *dynamic) checkByoNSG(ctx context.Context, subnetByID map[string]*mgmtnetwork.Subnet) (bool, error) {
```

Comment:
is ctx being used?

Reply:
Nope it's not.  I initially thought I would need a call to ARM, but turned out I didn't.  Should I just remove it?

Reply:
i think we should since there is no rpc call

Reply:
Removed accordingly.